cargo-optee/src/main.rs

```diff
@@ -0,0 +1,567 @@
+        &project_path,
+        component_type,
+        common.arch,
+        Some(common.debug),
+        None, // std not applicable for CA/Plugin
+        None, // ta_dev_kit_dir not needed for CA/Plugin
+        optee_client_export,
+        None, // signing_key not needed for CA/Plugin
+        uuid_path,
+    )?;
+
+    // Print the final configuration being used
+    build_config.print_config(component_type, &project_path);
+
+    // Get required optee_client_export and resolve relative to project
+    let optee_client_export_config = build_config.require_optee_client_export()?;
+    let optee_client_export =
+        resolve_path_relative_to_project(&optee_client_export_config, &project_path);
+
+    // Validate that optee_client_export exists (print absolute path)
+    if !optee_client_export.exists() {
+        bail!(
+            "OP-TEE client export directory does not exist: {:?}",
+            optee_client_export
+        );
+    }
+
+    // Merge env variables: CLI overrides + metadata env
+    let mut merged_env = build_config.env.clone();
+    merged_env.extend(common.env);
+
+    let ca_config = ca_builder::CaBuildConfig {
+        arch: build_config.arch,
+        optee_client_export,
+        debug: build_config.debug,
+        path: project_path,
+        plugin,
+        uuid_path: if plugin {
+            build_config.uuid_path.clone()
+        } else {
+            None
+        },
+        env: merged_env,
+        no_default_features: common.no_default_features,
+        features: common.features,
+    };
+
+    ca_builder::build_ca(ca_config, install_target_dir.map(|p| p.as_path()))
+}
+
+/// Common build command arguments shared across TA, CA, and Plugin builds
+#[derive(Debug, Args)]
+struct CommonBuildArgs {
+    /// Path to the Cargo.toml manifest file
+    #[arg(long = "manifest-path")]
+    manifest_path: Option<PathBuf>,
+
+    /// Target architecture (default: aarch64)
+    #[arg(long = "arch")]
+    arch: Option<Arch>,
+
+    /// Enable debug build (default: false)
+    #[arg(long = "debug")]
+    debug: bool,
+
+    /// Environment overrides in the form of `"KEY=VALUE"` strings. This flag can be repeated.
+    ///
+    /// This is generally not needed to be used explicitly during regular development.
+    ///
+    /// This makes sense to be used to specify custom var e.g. `RUSTFLAGS`.
+    #[arg(long = "env", value_parser = parse_env_var, action = clap::ArgAction::Append)]
+    env: Vec<(String, String)>,
+
+    /// Disable default features (will append --no-default-features to cargo build)
+    #[arg(long = "no-default-features")]
+    no_default_features: bool,
+
+    /// Custom features to enable (will append --features to cargo build)
+    #[arg(long = "features")]
+    features: Option<String>,
+}
+
+#[derive(Debug, Parser)]
+#[clap(version = env!("CARGO_PKG_VERSION"))]
+#[clap(about = "Build tool for OP-TEE Rust projects")]
+pub(crate) struct Cli {
+    #[clap(subcommand)]
+    cmd: Command,
+}
```

Comment:
Recommend to move all cli structs definition to another file `cli.rs` to make the `main.rs` simple and clean